ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/snapsync/SnapWorldStateDownloadProcess.java

```diff
@@ -320,8 +319,10 @@ public SnapWorldStateDownloadProcess build() {
                   "checkNewPivotBlock",
                   tasks -> {
                     pivotBlockManager.check(
-                        blockHeader -> {
-                          if (snapSyncState.isHealInProgress()) downloadState.clearTrieNodes();
+                        (blockHeader, newBlockFound) -> {
```

Comment:
same here, introduce a variable or method for this action to perform on pivot switch for readibility and remove code duplication

Reply:
done